pages/api/upload.ts

```diff
@@ -0,0 +1,57 @@
+import { NextApiRequest, NextApiResponse } from 'next'
+import { PrismaClient, transactions } from '@prisma/client'
+import TransactionImport from '../model/transactionImport'
+
+
+const prisma = new PrismaClient()
+
+const checkacnt = async (accountId : string)=>{
+    console.log('Finding: ', accountId)
+
+    let acc = await prisma.accounts.findUnique({
+        where:{ account_number: accountId}
+    })
+
+    if (! acc) {
+        acc = await prisma.accounts.create({
+            data:{
+                account_number: accountId,
+            }
+         })      
+    }  
+    
+    return acc.id
+}
+
+export default async (req : NextApiRequest, res : NextApiResponse) => {    
+    if(req.method !=='POST'){
+        res.status(405).json({})
+    }
+
+    const body : TransactionImport = JSON.parse(req.body);
+    
+    if (body) {
+        const accountId = await checkacnt(body.accountNumber)
+
+        const transactions  = body.transactions.map((ofx : any) => {
+            const dt = new Date(ofx['DTPOSTED'].substring(0,4)+"-"+ofx['DTPOSTED'].substring(4,6)+"-"+ofx['DTPOSTED'].substring(6,8))
```

Comment:
It might be better to have its own function for transforming the date from a string to a Date object. That way it can be easier to see the logic and the intent for the code.